### PR TITLE
Rework action code

### PR DESCRIPTION
**Pull request by ChrisChinchilla**

### CODE REVIEW

**Review by jdkato — CHANGES_REQUESTED** [review]

File: src/features/vsProvider.ts [hunk 1]
```diff
@@ -144,43 +147,46 @@ export default class ValeProvider implements vscode.CodeActionProvider {
...
-        suggestion,
-        alert.Action.Name,
-      ],
-    };
-
-    actions.push(action);
-    return actions; */
+    // TODO: Handle spelling, for now check we are not handling anything but replace or remove and so don't return empty actions.
+    // Also currently handles rules with no actions defined, so again doesn't return empty actions
+    if (alert.Action.Name !== "suggest" && alert.Action.Params !== null) {
```
[the remark this answers is not in the document]
Comment (ChrisChinchilla) [reply src/features/vsProvider.ts:155]: @jdkato Hmm, I'm confused, I checked `remove` actions and they worked, I was trying to find a way to handle the (many) rules that have no actions defined, maybe I could check for a null name instead I guess.

File: src/features/vsProvider.ts [hunk 2]
```diff
@@ -144,43 +147,46 @@ export default class ValeProvider implements vscode.CodeActionProvider {
     let diagnostic: vscode.Diagnostic = context.diagnostics[0];
     let actions: vscode.CodeAction[] = [];
 
-    return actions;
-
-    /* TODO: This needs more work / testing.
-    if (diagnostic === undefined) {
```
[the remark this answers is not in the document]
Comment (ChrisChinchilla) [reply src/features/vsProvider.ts:150]: @jdkato I am not familiar with that error, any reference so I can test it?

[the remark this answers is not in the document]
Comment (jdkato) [reply src/features/vsProvider.ts:150]: It's the one you mentioned in the Slack DM. You should see it in the developer console.

File: src/features/vsProvider.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdkato) [reply src/features/vsProvider.ts:155]: Something like:

```yaml
action:
  name: remove
```

**Comment by ChrisChinchilla** [comment]
OK @jdkato Try the latest changes. it also made me realise that a seperate change is to go through a lot of the older rules and add actions to them. Once this is merged I'll also look at spelling.

**Review by jdkato — APPROVED** [review]

**Comment by jdkato** [comment]
Looks good. I'm going to test it a few days before I make a release, just to be sure.

**Comment by ChrisChinchilla** [comment]
@jdkato Sure, we can even squeeze some other features in to a release too. I'l' also slowly start updating rules with actions.

**Pull request merged**